server/routerlicious/packages/routerlicious-base/src/utils/runnerHelper.ts

```diff
@@ -0,0 +1,49 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { serializeError } from "serialize-error";
+import { IWebServer } from "@fluidframework/server-services-core";
+import { Lumber } from "@fluidframework/server-services-telemetry";
+import { promiseTimeout } from "@fluidframework/server-services-client";
+import { Deferred } from "@fluidframework/common-utils";
+
+export async function runnerHttpServerStop(
+	server: IWebServer,
+	runningDeferredPromise: Deferred<void>,
+	runnerServerCloseTimeoutMs: number,
+	runnerMetric: Lumber,
+	caller: string | undefined,
+	uncaughtException: any | undefined,
+): Promise<void> {
+	try {
+		// Close the underlying server and then resolve the runner once closed
+		await promiseTimeout(runnerServerCloseTimeoutMs, server.close());
+		if (caller === "uncaughtException") {
+			runningDeferredPromise?.reject({
+				uncaughtException: serializeError(uncaughtException),
+			}); // reject the promise so that the runService exits the process with exit(1)
+		} else {
+			runningDeferredPromise?.resolve();
+		}
+		if (!runnerMetric.isCompleted()) {
+			runnerMetric.success(`${runnerMetric.eventName} stopped`);
+		}
```

Comment:
Maybe add a warning when metric already completed? Just for debugging purpose and I don't think this would have a lot of logs

Reply:
updated